### PR TITLE
Adding Network Icon and make it save settings

### PR DESCRIPTION
**Pull request by piyalbasu**
Screen Shot 2021-04-07 at 7 42 00 PM
Screen Shot 2021-04-07 at 7 42 20 PM
Screen Shot 2021-04-07 at 7 42 11 PM

### CODE REVIEW

File: @shared/api/helpers/extensionMessaging.ts [hunk 1]
```diff
@@ -14,7 +14,12 @@ declare global {
 }
 
 export const sendMessageToContentScript = (msg: {}): Promise<Response> => {
-  const MESSAGE_ID = Date.now();
+  /* 
+    In the case of multiple calls coming in sequentially, we use this MESSAGE_ID to make sure we're responding to
+    the appropriate message sender. Otherwise, we can run into race conditions where we simply resolve all 
+    sent messages with the first thing that comes back.
+  */
+  const MESSAGE_ID = Date.now() + Math.random();
```
Comment (piyalbasu) [inline @shared/api/helpers/extensionMessaging.ts:22]: Weird race condition in dev I never noticed before

Comment (piyalbasu) [reply @shared/api/helpers/extensionMessaging.ts:22]: Not an issue in prod

File: @shared/helpers/stellar.ts [hunk 2]
```diff
@@ -23,32 +24,13 @@ export const MAINNET_NETWORK_DETAILS = {
 } as NetworkDetails;
 
 export const TESTNET_NETWORK_DETAILS = {
+  isTestnet: true,
```
Comment (piyalbasu) [inline @shared/helpers/stellar.ts:27]: add these here so we can also statically import these complete configs if we need to

File: extension/webpack.common.js [hunk 3]
```diff
@@ -83,9 +83,6 @@ const commonConfig = (env = { EXPERIMENTAL: false, PRODUCTION: false }) => ({
     new webpack.DefinePlugin({
       EXPERIMENTAL: env.EXPERIMENTAL,
     }),
-    new webpack.DefinePlugin({
-      PRODUCTION: env.PRODUCTION,
-    }),
```
Comment (piyalbasu) [inline extension/webpack.common.js:88]: no longer needed - this was used to toggle test/main net

File: extension/webpack.extension.js [hunk 4]
```diff
@@ -2,8 +2,12 @@ const merge = require("webpack-merge");
 const webpack = require("webpack");
 const { commonConfig } = require("./webpack.common.js");
 
-const prodConfig = {
+const prodConfig = (env = { PRODUCTION: false }) => ({
   mode: "production",
+  optimization: {
+    minimize: env.PRODUCTION,
+  },
```
Comment (piyalbasu) [inline extension/webpack.extension.js:9]: semi-unrelated, but was useful for debugging the race condition in dev

File: extension/package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "build": "webpack --config webpack.extension.js",
     "build:experimental": "yarn build --env.EXPERIMENTAL=true",
-    "build:production": "yarn build --env.PRODUCTION=true",
+    "build:production": "yarn build --env.PRODUCTION",
```
Comment (piyalbasu) [inline extension/package.json:9]: PRODUCTION must be set this way to be cast as a boolean instead of a string

**Review by quietbits — APPROVED** [review]

**Pull request merged**